CI is failing

With recent merges, the CI is now failing (where it did not fail in the PR's branch).

One fail, in the [Py3.8 build](https://github.com/bluesky/hklpy/runs/3192758925?check_suite_focus=true), is due to missing `pandoc` package.  This fail blocks building of the documentation.

```sh
  File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/nbconvert/utils/pandoc.py", line 77, in get_pandoc_version
    raise PandocMissing()
nbconvert.utils.pandoc.PandocMissing: Pandoc wasn't found.
Please check that pandoc is installed:
https://pandoc.org/installing.html
```

Another failure, in the [Py3.6 build](https://github.com/bluesky/hklpy/runs/3188404665?check_suite_focus=true), is due to a problem with the `xarray` package.

```sh
/usr/share/miniconda/envs/test-hklpy/lib/python3.6/site-packages/xarray/core/dataarray.py:24: in <module>
    from . import (
E     File "/usr/share/miniconda/envs/test-hklpy/lib/python3.6/site-packages/xarray/core/computation.py", line 4
E       from __future__ import annotations
E                                        ^
E   SyntaxError: future feature annotations is not defined
```

Here's the full output from `pytest -vvv .`:

<details>

```sh
Step 15/15 : RUN pytest -vvv .
 ---> Running in 31a8bf51cf8d
============================= test session starts ==============================
platform linux -- Python 3.6.13, pytest-6.2.4, py-1.10.0, pluggy-0.13.1 -- /opt/conda/envs/py36/bin/python
cachedir: .pytest_cache
rootdir: /opt/hklpy, configfile: pytest.ini
plugins: cov-2.12.1
collecting ... collected 0 items / 9 errors

==================================== ERRORS ====================================
_________________ ERROR collecting hkl/tests/test_diffract.py __________________
../conda/envs/py36/lib/python3.6/site-packages/_pytest/python.py:578: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../conda/envs/py36/lib/python3.6/site-packages/_pytest/pathlib.py:524: in import_path
    importlib.import_module(module_name)
../conda/envs/py36/lib/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:665: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:678: in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
hkl/__init__.py:24: in <module>
    from .geometries import (  # noqa: F401, F402, E402
hkl/geometries.py:43: in <module>
    from .diffract import Diffractometer
hkl/diffract.py:22: in <module>
    import pint
../conda/envs/py36/lib/python3.6/site-packages/pint/__init__.py:14: in <module>
    from .context import Context
../conda/envs/py36/lib/python3.6/site-packages/pint/context.py:15: in <module>
    from .definitions import Definition, UnitDefinition
../conda/envs/py36/lib/python3.6/site-packages/pint/definitions.py:13: in <module>
    from .converters import LogarithmicConverter, OffsetConverter, ScaleConverter
../conda/envs/py36/lib/python3.6/site-packages/pint/converters.py:12: in <module>
    from .compat import HAS_NUMPY, exp, log  # noqa: F401
../conda/envs/py36/lib/python3.6/site-packages/pint/compat.py:163: in <module>
    from xarray import DataArray, Dataset, Variable
../conda/envs/py36/lib/python3.6/site-packages/xarray/__init__.py:3: in <module>
    from . import testing, tutorial, ufuncs
../conda/envs/py36/lib/python3.6/site-packages/xarray/testing.py:9: in <module>
    from xarray.core.dataarray import DataArray
../conda/envs/py36/lib/python3.6/site-packages/xarray/core/dataarray.py:24: in <module>
    from . import (
E     File "/opt/conda/envs/py36/lib/python3.6/site-packages/xarray/core/computation.py", line 4
E       from __future__ import annotations
E                                        ^
E   SyntaxError: future feature annotations is not defined
________________ ERROR collecting hkl/tests/test_extra_motor.py ________________
../conda/envs/py36/lib/python3.6/site-packages/_pytest/python.py:578: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../conda/envs/py36/lib/python3.6/site-packages/_pytest/pathlib.py:524: in import_path
    importlib.import_module(module_name)
../conda/envs/py36/lib/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:665: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:678: in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
hkl/__init__.py:24: in <module>
    from .geometries import (  # noqa: F401, F402, E402
hkl/geometries.py:43: in <module>
    from .diffract import Diffractometer
hkl/diffract.py:22: in <module>
    import pint
../conda/envs/py36/lib/python3.6/site-packages/pint/__init__.py:14: in <module>
    from .context import Context
../conda/envs/py36/lib/python3.6/site-packages/pint/context.py:15: in <module>
    from .definitions import Definition, UnitDefinition
../conda/envs/py36/lib/python3.6/site-packages/pint/definitions.py:13: in <module>
    from .converters import LogarithmicConverter, OffsetConverter, ScaleConverter
../conda/envs/py36/lib/python3.6/site-packages/pint/converters.py:12: in <module>
    from .compat import HAS_NUMPY, exp, log  # noqa: F401
../conda/envs/py36/lib/python3.6/site-packages/pint/compat.py:163: in <module>
    from xarray import DataArray, Dataset, Variable
../conda/envs/py36/lib/python3.6/site-packages/xarray/__init__.py:3: in <module>
    from . import testing, tutorial, ufuncs
../conda/envs/py36/lib/python3.6/site-packages/xarray/testing.py:9: in <module>
    from xarray.core.dataarray import DataArray
../conda/envs/py36/lib/python3.6/site-packages/xarray/core/dataarray.py:24: in <module>
    from . import (
E     File "/opt/conda/envs/py36/lib/python3.6/site-packages/xarray/core/computation.py", line 4
E       from __future__ import annotations
E                                        ^
E   SyntaxError: future feature annotations is not defined
___________________ ERROR collecting hkl/tests/test_fourc.py ___________________
../conda/envs/py36/lib/python3.6/site-packages/_pytest/python.py:578: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../conda/envs/py36/lib/python3.6/site-packages/_pytest/pathlib.py:524: in import_path
    importlib.import_module(module_name)
../conda/envs/py36/lib/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:665: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:678: in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
hkl/__init__.py:24: in <module>
    from .geometries import (  # noqa: F401, F402, E402
hkl/geometries.py:43: in <module>
    from .diffract import Diffractometer
hkl/diffract.py:22: in <module>
    import pint
../conda/envs/py36/lib/python3.6/site-packages/pint/__init__.py:14: in <module>
    from .context import Context
../conda/envs/py36/lib/python3.6/site-packages/pint/context.py:15: in <module>
    from .definitions import Definition, UnitDefinition
../conda/envs/py36/lib/python3.6/site-packages/pint/definitions.py:13: in <module>
    from .converters import LogarithmicConverter, OffsetConverter, ScaleConverter
../conda/envs/py36/lib/python3.6/site-packages/pint/converters.py:12: in <module>
    from .compat import HAS_NUMPY, exp, log  # noqa: F401
../conda/envs/py36/lib/python3.6/site-packages/pint/compat.py:163: in <module>
    from xarray import DataArray, Dataset, Variable
../conda/envs/py36/lib/python3.6/site-packages/xarray/__init__.py:3: in <module>
    from . import testing, tutorial, ufuncs
../conda/envs/py36/lib/python3.6/site-packages/xarray/testing.py:9: in <module>
    from xarray.core.dataarray import DataArray
../conda/envs/py36/lib/python3.6/site-packages/xarray/core/dataarray.py:24: in <module>
    from . import (
E     File "/opt/conda/envs/py36/lib/python3.6/site-packages/xarray/core/computation.py", line 4
E       from __future__ import annotations
E                                        ^
E   SyntaxError: future feature annotations is not defined
___________________ ERROR collecting hkl/tests/test_init.py ____________________
../conda/envs/py36/lib/python3.6/site-packages/_pytest/python.py:578: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../conda/envs/py36/lib/python3.6/site-packages/_pytest/pathlib.py:524: in import_path
    importlib.import_module(module_name)
../conda/envs/py36/lib/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:665: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:678: in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
hkl/__init__.py:24: in <module>
    from .geometries import (  # noqa: F401, F402, E402
hkl/geometries.py:43: in <module>
    from .diffract import Diffractometer
hkl/diffract.py:22: in <module>
    import pint
../conda/envs/py36/lib/python3.6/site-packages/pint/__init__.py:14: in <module>
    from .context import Context
../conda/envs/py36/lib/python3.6/site-packages/pint/context.py:15: in <module>
    from .definitions import Definition, UnitDefinition
../conda/envs/py36/lib/python3.6/site-packages/pint/definitions.py:13: in <module>
    from .converters import LogarithmicConverter, OffsetConverter, ScaleConverter
../conda/envs/py36/lib/python3.6/site-packages/pint/converters.py:12: in <module>
    from .compat import HAS_NUMPY, exp, log  # noqa: F401
../conda/envs/py36/lib/python3.6/site-packages/pint/compat.py:163: in <module>
    from xarray import DataArray, Dataset, Variable
../conda/envs/py36/lib/python3.6/site-packages/xarray/__init__.py:3: in <module>
    from . import testing, tutorial, ufuncs
../conda/envs/py36/lib/python3.6/site-packages/xarray/testing.py:9: in <module>
    from xarray.core.dataarray import DataArray
../conda/envs/py36/lib/python3.6/site-packages/xarray/core/dataarray.py:24: in <module>
    from . import (
E     File "/opt/conda/envs/py36/lib/python3.6/site-packages/xarray/core/computation.py", line 4
E       from __future__ import annotations
E                                        ^
E   SyntaxError: future feature annotations is not defined
__________________ ERROR collecting hkl/tests/test_sample.py ___________________
../conda/envs/py36/lib/python3.6/site-packages/_pytest/python.py:578: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../conda/envs/py36/lib/python3.6/site-packages/_pytest/pathlib.py:524: in import_path
    importlib.import_module(module_name)
../conda/envs/py36/lib/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:665: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:678: in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
hkl/__init__.py:24: in <module>
    from .geometries import (  # noqa: F401, F402, E402
hkl/geometries.py:43: in <module>
    from .diffract import Diffractometer
hkl/diffract.py:22: in <module>
    import pint
../conda/envs/py36/lib/python3.6/site-packages/pint/__init__.py:14: in <module>
    from .context import Context
../conda/envs/py36/lib/python3.6/site-packages/pint/context.py:15: in <module>
    from .definitions import Definition, UnitDefinition
../conda/envs/py36/lib/python3.6/site-packages/pint/definitions.py:13: in <module>
    from .converters import LogarithmicConverter, OffsetConverter, ScaleConverter
../conda/envs/py36/lib/python3.6/site-packages/pint/converters.py:12: in <module>
    from .compat import HAS_NUMPY, exp, log  # noqa: F401
../conda/envs/py36/lib/python3.6/site-packages/pint/compat.py:163: in <module>
    from xarray import DataArray, Dataset, Variable
../conda/envs/py36/lib/python3.6/site-packages/xarray/__init__.py:3: in <module>
    from . import testing, tutorial, ufuncs
../conda/envs/py36/lib/python3.6/site-packages/xarray/testing.py:9: in <module>
    from xarray.core.dataarray import DataArray
../conda/envs/py36/lib/python3.6/site-packages/xarray/core/dataarray.py:24: in <module>
    from . import (
E     File "/opt/conda/envs/py36/lib/python3.6/site-packages/xarray/core/computation.py", line 4
E       from __future__ import annotations
E                                        ^
E   SyntaxError: future feature annotations is not defined
______________ ERROR collecting hkl/tests/test_save_restore_UB.py ______________
../conda/envs/py36/lib/python3.6/site-packages/_pytest/python.py:578: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../conda/envs/py36/lib/python3.6/site-packages/_pytest/pathlib.py:524: in import_path
    importlib.import_module(module_name)
../conda/envs/py36/lib/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:665: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:678: in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
hkl/__init__.py:24: in <module>
    from .geometries import (  # noqa: F401, F402, E402
hkl/geometries.py:43: in <module>
    from .diffract import Diffractometer
hkl/diffract.py:22: in <module>
    import pint
../conda/envs/py36/lib/python3.6/site-packages/pint/__init__.py:14: in <module>
    from .context import Context
../conda/envs/py36/lib/python3.6/site-packages/pint/context.py:15: in <module>
    from .definitions import Definition, UnitDefinition
../conda/envs/py36/lib/python3.6/site-packages/pint/definitions.py:13: in <module>
    from .converters import LogarithmicConverter, OffsetConverter, ScaleConverter
../conda/envs/py36/lib/python3.6/site-packages/pint/converters.py:12: in <module>
    from .compat import HAS_NUMPY, exp, log  # noqa: F401
../conda/envs/py36/lib/python3.6/site-packages/pint/compat.py:163: in <module>
    from xarray import DataArray, Dataset, Variable
../conda/envs/py36/lib/python3.6/site-packages/xarray/__init__.py:3: in <module>
    from . import testing, tutorial, ufuncs
../conda/envs/py36/lib/python3.6/site-packages/xarray/testing.py:9: in <module>
    from xarray.core.dataarray import DataArray
../conda/envs/py36/lib/python3.6/site-packages/xarray/core/dataarray.py:24: in <module>
    from . import (
E     File "/opt/conda/envs/py36/lib/python3.6/site-packages/xarray/core/computation.py", line 4
E       from __future__ import annotations
E                                        ^
E   SyntaxError: future feature annotations is not defined
__________________ ERROR collecting hkl/tests/test_tardis.py ___________________
../conda/envs/py36/lib/python3.6/site-packages/_pytest/python.py:578: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../conda/envs/py36/lib/python3.6/site-packages/_pytest/pathlib.py:524: in import_path
    importlib.import_module(module_name)
../conda/envs/py36/lib/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:665: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:678: in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
hkl/__init__.py:24: in <module>
    from .geometries import (  # noqa: F401, F402, E402
hkl/geometries.py:43: in <module>
    from .diffract import Diffractometer
hkl/diffract.py:22: in <module>
    import pint
../conda/envs/py36/lib/python3.6/site-packages/pint/__init__.py:14: in <module>
    from .context import Context
../conda/envs/py36/lib/python3.6/site-packages/pint/context.py:15: in <module>
    from .definitions import Definition, UnitDefinition
../conda/envs/py36/lib/python3.6/site-packages/pint/definitions.py:13: in <module>
    from .converters import LogarithmicConverter, OffsetConverter, ScaleConverter
../conda/envs/py36/lib/python3.6/site-packages/pint/converters.py:12: in <module>
    from .compat import HAS_NUMPY, exp, log  # noqa: F401
../conda/envs/py36/lib/python3.6/site-packages/pint/compat.py:163: in <module>
    from xarray import DataArray, Dataset, Variable
../conda/envs/py36/lib/python3.6/site-packages/xarray/__init__.py:3: in <module>
    from . import testing, tutorial, ufuncs
../conda/envs/py36/lib/python3.6/site-packages/xarray/testing.py:9: in <module>
    from xarray.core.dataarray import DataArray
../conda/envs/py36/lib/python3.6/site-packages/xarray/core/dataarray.py:24: in <module>
    from . import (
E     File "/opt/conda/envs/py36/lib/python3.6/site-packages/xarray/core/computation.py", line 4
E       from __future__ import annotations
E                                        ^
E   SyntaxError: future feature annotations is not defined
___________________ ERROR collecting hkl/tests/test_user.py ____________________
../conda/envs/py36/lib/python3.6/site-packages/_pytest/python.py:578: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../conda/envs/py36/lib/python3.6/site-packages/_pytest/pathlib.py:524: in import_path
    importlib.import_module(module_name)
../conda/envs/py36/lib/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:665: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:678: in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
hkl/__init__.py:24: in <module>
    from .geometries import (  # noqa: F401, F402, E402
hkl/geometries.py:43: in <module>
    from .diffract import Diffractometer
hkl/diffract.py:22: in <module>
    import pint
../conda/envs/py36/lib/python3.6/site-packages/pint/__init__.py:14: in <module>
    from .context import Context
../conda/envs/py36/lib/python3.6/site-packages/pint/context.py:15: in <module>
    from .definitions import Definition, UnitDefinition
../conda/envs/py36/lib/python3.6/site-packages/pint/definitions.py:13: in <module>
    from .converters import LogarithmicConverter, OffsetConverter, ScaleConverter
../conda/envs/py36/lib/python3.6/site-packages/pint/converters.py:12: in <module>
    from .compat import HAS_NUMPY, exp, log  # noqa: F401
../conda/envs/py36/lib/python3.6/site-packages/pint/compat.py:163: in <module>
    from xarray import DataArray, Dataset, Variable
../conda/envs/py36/lib/python3.6/site-packages/xarray/__init__.py:3: in <module>
    from . import testing, tutorial, ufuncs
../conda/envs/py36/lib/python3.6/site-packages/xarray/testing.py:9: in <module>
    from xarray.core.dataarray import DataArray
../conda/envs/py36/lib/python3.6/site-packages/xarray/core/dataarray.py:24: in <module>
    from . import (
E     File "/opt/conda/envs/py36/lib/python3.6/site-packages/xarray/core/computation.py", line 4
E       from __future__ import annotations
E                                        ^
E   SyntaxError: future feature annotations is not defined
___________________ ERROR collecting hkl/tests/test_util.py ____________________
../conda/envs/py36/lib/python3.6/site-packages/_pytest/python.py:578: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
../conda/envs/py36/lib/python3.6/site-packages/_pytest/pathlib.py:524: in import_path
    importlib.import_module(module_name)
../conda/envs/py36/lib/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:941: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:665: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:678: in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
hkl/__init__.py:24: in <module>
    from .geometries import (  # noqa: F401, F402, E402
hkl/geometries.py:43: in <module>
    from .diffract import Diffractometer
hkl/diffract.py:22: in <module>
    import pint
../conda/envs/py36/lib/python3.6/site-packages/pint/__init__.py:14: in <module>
    from .context import Context
../conda/envs/py36/lib/python3.6/site-packages/pint/context.py:15: in <module>
    from .definitions import Definition, UnitDefinition
../conda/envs/py36/lib/python3.6/site-packages/pint/definitions.py:13: in <module>
    from .converters import LogarithmicConverter, OffsetConverter, ScaleConverter
../conda/envs/py36/lib/python3.6/site-packages/pint/converters.py:12: in <module>
    from .compat import HAS_NUMPY, exp, log  # noqa: F401
../conda/envs/py36/lib/python3.6/site-packages/pint/compat.py:163: in <module>
    from xarray import DataArray, Dataset, Variable
../conda/envs/py36/lib/python3.6/site-packages/xarray/__init__.py:3: in <module>
    from . import testing, tutorial, ufuncs
../conda/envs/py36/lib/python3.6/site-packages/xarray/testing.py:9: in <module>
    from xarray.core.dataarray import DataArray
../conda/envs/py36/lib/python3.6/site-packages/xarray/core/dataarray.py:24: in <module>
    from . import (
E     File "/opt/conda/envs/py36/lib/python3.6/site-packages/xarray/core/computation.py", line 4
E       from __future__ import annotations
E                                        ^
E   SyntaxError: future feature annotations is not defined
=========================== short test summary info ============================
ERROR hkl/tests/test_diffract.py
ERROR hkl/tests/test_extra_motor.py
ERROR hkl/tests/test_fourc.py
ERROR hkl/tests/test_init.py
ERROR hkl/tests/test_sample.py
ERROR hkl/tests/test_save_restore_UB.py
ERROR hkl/tests/test_tardis.py
ERROR hkl/tests/test_user.py
ERROR hkl/tests/test_util.py
!!!!!!!!!!!!!!!!!!! Interrupted: 9 errors during collection !!!!!!!!!!!!!!!!!!!!
============================== 9 errors in 2.68s ===============================
```

</details>

--- a/.github/tmate.yml
+++ b/.github/tmate.yml
@@ -15,7 +15,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        # python-version: [3.6, 3.7, 3.8]
+        # python-version: [3.7, 3.8, 3.9]
         python-version: [3.8]
 
     steps:

--- a/.github/workflows/conda_unit_test.yml
+++ b/.github/workflows/conda_unit_test.yml
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: [3.7, 3.8, 3.9]
 
     steps:
     - uses: actions/checkout@v2

--- a/.github/workflows/publish-docs.yml
+++ b/.github/workflows/publish-docs.yml
@@ -56,7 +56,7 @@ jobs:
     - name: Install publishing requirements
       shell: bash -l {0}
       run: |
-        conda install jupyter nbconvert sphinx sphinxcontrib-napoleon -c defaults -c conda-forge
+        conda install jupyter nbconvert sphinx sphinxcontrib-napoleon make flake8 black pandoc -c defaults -c conda-forge
         pip install sphinx-rtd-theme
 
     - name: Install the package locally
@@ -71,8 +71,9 @@ jobs:
         # conda list jupyter
         # conda list nbconvert
         env | sort | grep -i CONDA
-        make -C examples/
-        make -C docs/ html
+        # make -C examples/
+        # make -C docs/ html
+        make
 
     - name: Deploy documentation to blueskyproject.io.
       # We pin to the SHA, not the tag, for security reasons.

--- a/RELEASE_NOTES.rst
+++ b/RELEASE_NOTES.rst
@@ -18,6 +18,11 @@ release date.
 
 .. [#] https://github.com/bluesky/hklpy/milestone/5
 
+Breaking Changes
+----------------
+
+* Python 3.6 no longer supported.
+
 v0.3.16 (2021-04-28)
 ================================
 

--- a/environment.yml
+++ b/environment.yml
@@ -5,7 +5,7 @@ channels:
   - defaults
   - aps-anl-tag
 dependencies:
-  - python >=3.6
+  - python >=3.7
   - bluesky >=1.6.7
   - hkl
   - ipython

